### ISSUE TITLE
Build Out Procedure Changing Slightly

### ISSUE THREAD

**Issue by sylus** [wet-boew/wet-boew-drupal#388]
Currently right now the build is accomplished by running a command similar to the following:

```
drush make --prepare-install --no-gitinfofile --working-copy https://raw.github.com/wet-boew/wet-boew-drupal/master/build-wetkit.make drupal_wet
```

Where the build-wetkit.make has the following contents:

```
includes[] = https://raw.github.com/wet-boew/wet-boew-drupal/master/drupal-org-core.make

; Add wetkit to the full Drupal distro build
projects[wetkit][type] = profile
projects[wetkit][download][type] = git
projects[wetkit][download][url] = https://github.com/wet-boew/wet-boew-drupal.git
projects[wetkit][download][revision] = master
```

The problem with this is when someone performs a pull request and travis does a build it isn't taking into account the code from the pull request as you can see it is using the wet-boew-drupal repo. So only when you merge will it then test the new code which kind of defeats the purpose.

I am proposing the new build out procedure is the following:

```
1) In your web root (or create a directory such as drupal_wet)
2) git clone https://github.com/wet-boew/wet-boew-drupal.git profiles/wetkit;
3) drush make --prepare-install profiles/wetkit/build-wetkit.make --yes
```

With the contents of build-wetkit.make having only relative includes:

```
api = 2
core = 7.x

projects[drupal][version] = 7.15
includes[] = drupal-org-core.make
includes[] = drupal-org.make
```

This would allow Travis to perform tests on every pull request that are valid ensuring we don't merge until Travis passes.

@joel-osc @joejoseph00 @thriuin @mgifford @eleclerc @openplus @StephenOTT

**Comment by eleclerc** [comment]
Totally agree, I've been running my setup in a similar way for a few weeks without any issues.

**Comment by patcon** [comment]
:+1: x infinity

Changes like this will really help increase build reproducibility. We're confined to operating at branch-heads only otherwise, which essentially means the install profile will only ever build the newest commits reliably, and all previous ones might break -- which is bad mojo :)

**Comment by joejoseph00** [comment]
I updated and enhanced this bash script to use this new method.  
http://wet.7pro.ca/sites/wet.7pro.ca/files/wet_drush_sh.txt
N.B: press CTRL-F5 for a forced refresh if you previously clicked on this link

usage is as follows:
./wet_drush.sh wetboewX h.simpson@fox.com mysql_user mysql_host mysql_pass drupal_pass "City of Springfield Web"

if you forget, just run it ./wet_drush.sh
and the parameters are verified one by one

It is designed to install on LXMP stacks that have UserDir mod enabled , installs to /home/username/public_html/x

to get to work on your distro, might have to change the Webserver user and group at the top of the script.

Aside from that, it automatically configures .htaccess , automatically configures folder permissions so that "install new module" works using the gui

has some validation and some basic error handling.

requires linux , and bash

**Comment by patcon** [comment]
Looks awesome @joejoseph00! Seems like a good time to add a comment for standardizing our project scripts (which I believe should be versioning controlled with the project (ie. profile) itself.

I'll start a new issue to discuss, as it's honestly separate from this :)

**Comment by lpgascon** [comment]
So has the procedure to Update the site been updated as well?

**Comment by sylus** [comment]
Procedure to update is pretty much the same just when you do a git clone ensure it is to profiles/wetkit
